core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -451,6 +451,15 @@ protected void doRun() {
                 }
                 performAction(node, transportPrimaryAction, true);
             } else {
+                if (state.version() < request.routedBasedOnClusterVersion()) {
+                    logger.trace("cluster state [{}] on node [{}] should have minimum version [{}] but was [{}], scheduling a retry: action [{}], request [{}]", request.shardId(), primary.currentNodeId(), request.routedBasedOnClusterVersion(), state.version(), actionName, request);
+                    retryBecauseUnavailable(request.shardId(), "expected minimum cluster state version " + request.routedBasedOnClusterVersion() + " but was " + state.version());
```

Comment:
maybe - failed to find primary but cluster state version [{}] is stale (expected at least [{}]).